group-coordinator/src/main/java/org/apache/kafka/coordinator/group/streams/assignor/StickyTaskAssignor.java

```
@@ -0,0 +1,401 @@
+        }
+        return memberWithLeastLoad(processWithLeastLoad.get());
+    }
+
+    private Member findMemberWithLeastLoad(final TaskId taskId) {
+        Set<Member> allMembers = processIdToState.entrySet().stream()
+            .flatMap(entry -> entry.getValue().memberToTaskCounts().keySet().stream()
+                .map(memberId -> new Member(entry.getKey(), memberId)))
+            .collect(Collectors.toSet());
+        return findMemberWithLeastLoad(allMembers, taskId, false);
+    }
+
+    private Member findMemberWithLeastLoad(final TaskId taskId, final Set<String> processes) {
+        Set<Member> allMembers = processes.stream()
+            .flatMap(processId -> processIdToState.get(processId).memberToTaskCounts().keySet().stream()
+                .map(memberId -> new Member(processId, memberId)))
+            .collect(Collectors.toSet());
+        return findMemberWithLeastLoad(allMembers, taskId, false);
+    }
+
+    private Member memberWithLeastLoad(final ProcessState processWithLeastLoad) {
+        Optional<String> memberWithLeastLoad = processWithLeastLoad.memberToTaskCounts().entrySet().stream()
+            .min(Map.Entry.comparingByValue())
+            .map(Map.Entry::getKey);
+        return memberWithLeastLoad.map(memberId -> new Member(processWithLeastLoad.processId(), memberId)).orElse(null);
+    }
+
+    private boolean hasUnfulfilledQuota(final Member member) {
+        return processIdToState.get(member.processId).memberToTaskCounts().get(member.memberId) < tasksPerMember;
+    }
+
+    private void assignStandby(final Set<TaskId> standbyTasks, final int numStandbyReplicas) {
+        for (TaskId task : standbyTasks) {
+            for (int i = 0; i < numStandbyReplicas; i++) {
+                final Set<String> availableProcesses = findAllowedProcesses(task);
+                if (availableProcesses.isEmpty()) {
+                    log.warn("Unable to assign " + (numStandbyReplicas - i) +
```

Should we throw here instead of just a warning?